### PR TITLE
Support Julia 0.5-pre

### PR DESCRIPTION
**Pull request by omus**
Making LibCURL use the latest Julia 0.5 syntax.

### CODE REVIEW

File: src/LibCURL.jl [hunk 1]
```diff
@@ -2,15 +2,20 @@ isdefined(Base, :__precompile__) && __precompile__()
...
 include("lC_exports_h.jl")
 include("lC_common_h.jl")
 
-@unix_only const libcurl = "libcurl"
-@windows_only const libcurl = Pkg.dir("WinRPM","deps","usr","$(Sys.ARCH)-w64-mingw32","sys-root","mingw","bin","libcurl-4")
+const libcurl = if is_windows()
+    Pkg.dir("WinRPM","deps","usr","$(Sys.ARCH)-w64-mingw32","sys-root","mingw","bin","libcurl-4")
+else
+    "libcurl"
+end
```
Comment (omus) [inline src/LibCURL.jl]: Slight change to the logic here. Previously if not UNIX or Windows `libcurl` would be undefined.

**Comment by PallHaraldsson** [comment]
@omus you seems to only be making use of the new syntax.

I'm not sure if it will work or not; even if libcurl is now included with Julia 0.5?! Should this package then be turned into a no-op on Julia 0.5?

**Comment by omus** [comment]
Compat lets you use the new syntax on older versions of Julia

Is there also a Julia interface to libcurl in base? Maybe this package could use the libcurl library installed by 0.5? I feel that having a Julia interface to the libcurl package is still useful.

File: src/lC_common_h.jl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 macro c(ret_type, func, arg_types, lib)
-  local args_in = Any[ symbol(string('a',x)) for x in 1:length(arg_types.args) ]
+  local args_in = Any[Symbol(string('a',x)) for x in 1:length(arg_types.args)]
```
Comment (yuyichao) [inline src/lC_common_h.jl]: No `string`.

Comment (omus) [inline src/lC_common_h.jl]: Is this better?
`local args_in = [symbol('a',i) for i in eachindex(arg_types.args)]`

Comment (yuyichao) [inline src/lC_common_h.jl]: `Sybol`

**Comment by PallHaraldsson** [comment]
Great that 0.5 is supported (I assume it works..), see https://github.com/JuliaWeb/MbedTLS.jl/pull/66#issuecomment-240368937 for alternative way.

**Pull request merged**